AzureCore/Source/Pipeline/Policies/ContentDecodePolicy.swift

```diff
@@ -58,10 +58,11 @@ public class ContentDecodePolicy: NSObject, PipelineStageProtocol, XMLParserDele
         return nil
     }
 
-    public func onResponse(_ response: inout PipelineResponse) {
+    public func onResponse(_ response: PipelineResponse, then completion: (PipelineResponse) -> Void) {
         let stream = response.value(forKey: "stream") as? Bool ?? false
         guard stream == false else { return }
         guard let httpResponse = response.httpResponse else { return }
+        var returnResponse = response.copy()
```

Comment:
Not clear why we had to make a copy, can you add some comments around this?

Reply:
Essentially, it's because arguments are `let` constants unless you mark them `inout`, which is discouraged. Thus instead of this:
```
mutateFoo(&foo)
```

You would do this:
```
foo = mutateFoo(foo)
```

But if you try to mutate foo in the method body, you'll get an error. Thus you copy the object and return the modified object.